isis/src/base/objs/IsisAml/IsisAml.cpp

```diff
@@ -678,6 +678,65 @@ void IsisAml::GetFileName(const QString &paramName,
 }
 
 
+
+
+
+
+
+
+
+/**
+ * Retrieves of a value for a parameter of type "cubename".
+ *
+ * @param paramName The partial or full name of the parameter to be retrieved.
+ * @param extension A default extension to add if it does not already exist on
+ * the file name.  For example, "txt" will make /mydir/myfile into
+ * /mydir/myfile.txt
+ *
+ * @return The value of the parameter.
+ */
+QString IsisAml::GetCubeName(const QString &paramName, QString extension) const {
+
+  const IsisParameterData *param = ReturnParam(paramName);
+
+  if (param->type != "cube") {
+    QString message = "Parameter [" + paramName + "] is not a cubename.";
+    throw Isis::IException(Isis::IException::Programmer, message, _FILEINFO_);
+  }
+
+  QString value;
+  if (param->values.size() == 0) {
+    if (param->defaultValues.size() == 0) {
+      QString message = "Parameter [" + paramName + "] has no value.";
+      throw Isis::IException(Isis::IException::User, message, _FILEINFO_);
+    }
+    else {
+      value = param->defaultValues[0];
+    }
+  }
+  else {
+    value = param->values[0];
+  }
+
+  Isis::FileName name(value);
+  if (extension != "") name = name.addExtension(extension);
```

Comment:
Do we nee to worry about the user putting something like `to=blah+32-bit` and then this converting it to `to=blah+32-bit.cub`? Does the FileName class handle that for us?

Reply:
Yes it does for cubes where it is expecting attributes, however, I'll check to see what happens if it is just a file name and there is a +

Reply:
FYI, plus signs are forbidden inside cube names